MACInfo/projects/MACInfo/src/module.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+
+import logging
+
+from pineapple.modules import Module, Request
+import json
+import os
+import urllib.request
+import re
+
+
+module = Module('MACInfo', logging.DEBUG)
+
+OUI_FILE = '/etc/pineapple/ouis'
+ONLINE_URL = 'https://macvendors.co/api/'
+
+@module.handles_action('check_mac_online')
+def check_mac_online(request: Request):
+    mac = request.user_input.upper()
+    strip_mac = mac.replace(' ','')
+    if '-' in strip_mac:
+        strip_mac = mac.replace('-',':')
+    strip_mac = re.search("^[a-fA-F0-9]{2}([:\-]?[a-fA-F0-9]{2}){2,5}$",strip_mac)
+
+    if strip_mac:
+        module.logger.debug("MATCH FOUND")
+        module.logger.debug(mac_reg)
+        module.logger.debug(strip_mac)
+        response = urllib.request.urlopen(f'{ONLINE_URL}/{strip_mac}/JSON')
+        data = response.read()
+        output_json = json.loads(data)
+        jsonData = output_json["result"]
+        company = jsonData.get('company')
+        mac_prefix = jsonData.get('mac_prefix')
+        maccountry = jsonData.get('country')
+        if maccountry == None:
+            maccountry = "Country not found"
+        address = jsonData.get('address')
+        start_hex = jsonData.get('start_hex')
+        end_hex = jsonData.get('end_hex')
+        mactype = jsonData.get('type')
+        return{'company':company,'address':address,'maccountry':maccountry,'mac_prefix':mac_prefix,'start_hex':start_hex,'end_hex':end_hex,'mactype':mactype}
+    else:
+        module.logger.debug("Not a valid MAC address")
+        return{'nomac':'Not a valid MAC address'}
+
+@module.handles_action('check_mac')
+def check_mac(request: Request):
+    module.logger.debug("Opening file")
+    mac = request.user_input.upper()
+    nospace_mac = mac.replace(' ','')
+    if ':' in nospace_mac:
+        strip_mac = mac.replace(':','')
+    elif '-' in nospace_mac:
+        strip_mac = mac.replace('-','')
+
+    reg_mac = re.search("^[a-fA-F0-9]{2}([:\-]?[a-fA-F0-9]{2}){2,5}$",nospace_mac)
+
+    if reg_mac:
+        with open(OUI_FILE) as f:
```

Comment:
I would suggest opening the OUI file into a global variable as a JSON/dict instead of opening it per request, this way the lookup should be (marginally) quicker, as opposed to loading a large file in each time. 

What do you think?

Reply:
Definently, didn't think of that first and as you said it would definently improve speed, so will do.